### ISSUE TITLE
Support high uid/gid in tarball

### ISSUE THREAD

**Issue by seblu** [backuppc/backuppc#254]
Hello,

**TLDR;** BackupPC 4.3 perfectly support high uid/gid with rsync (backup and restore), but exporting a tar archive truncate the uid/gid, resulting in broken archive. **This is a feature request to support high uid/gid in tarball**. 

To give you some context, I recently needed to restore backups of a server running virtual containers managed by systemd-machined. This containers manager take advantages of linux kernel user/group namespaces to secure containers (a very good thing) and as a consequence, in the host filesystem, machines files have very high uid/gid. By high, the man page of  [systemd-nspawn](https://www.freedesktop.org/software/systemd/man/systemd-nspawn.html) says:
> randomly chosen between the host UID/GIDs of 524288 and 1878982656

I'm my case, the uid 0 in the container is the uid 156827648 on the host. All following container uid are shifted from this one, so uid 1 is 156827649, and so on.

When Backuppc 4.x rsync the content of the host server it stores correctly these high uid/gid in its own internal format, so direct restore works correctly. But when times to save it to a tar archive, the uid/gid are truncated. The truncation is not done in an helping way as LSB are dropped, so you cannot easily shift back these uid to something with meaning (in the context of containers shifting).

So, a possible solution is to export tarball with a modern tar format (like POSIX.1-2001). There is a synthetic table of each format limitation on the [GNU Tar manpage](https://www.gnu.org/software/tar/manual/html_chapter/tar_8.html).

**Comment by craigbarratt** [comment]
I added support for pac headers (posix.1-2001) via commit a3ccfc2, but it has only minimal testing.  Please try it out and see if it works.

**Comment by seblu** [comment]
So fast! :open_mouth: 

Here is a recap of tests I made so far:

### The uid/gid are correctly reported with **BackupPC_tarCreate**
```sh
$ BackupPC_tarCreate -h foo -n 42 -s / -L /var/lib/machines
   700       0/0           0 ./var/lib/machines/
   755 156827648/156827648           0 ./var/lib/machines/foo/
   755 156827648/156827648           0 ./var/lib/machines/foo/bin/
   755 156827648/156827648           0 ./var/lib/machines/foo/boot/
...
  755 156827648/156827648      771552 ./var/lib/machines/foo/bin/zsh5
...
   660 156827648/156827692           0 ./var/lib/machines/foo/dev/agpgart
   660 156827648/156827677           0 ./var/lib/machines/foo/dev/audio
...
   600 156827648/156827653           0 ./var/lib/machines/foo/dev/console
   777 156827648/156827648           0 ./var/lib/machines/foo/dev/core -> /proc/kcore
   660 156827648/156827677           0 ./var/lib/machines/foo/dev/dsp
```
### Tarball are correctly extracted by both *gnu tar* and *bsd tar*.
```sh
$ tar --version                                                                                          (2019-02-12 15:22:46 CET)
tar (GNU tar) 1.31
Copyright (C) 2019 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <https://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Written by John Gilmore and Jay Fenlason.
$ time fakeroot tar xf 
$ time fakeroot tar xf foo.tar
fakeroot tar xf foo.tar  13,97s user 128,04s system 16% cpu 13:59,38 total
$ echo $?
0
```
```sh
$ bsdtar --version
bsdtar 3.3.3 - libarchive 3.3.3 zlib/1.2.11 liblzma/5.2.4 bz2lib/1.0.6 liblz4/1.8.2 libzstd/1.3.5 
$ time fakeroot bsdtar xf foo.tar
fakeroot bsdtar xf foo.tar  28,68s user 170,85s system 16% cpu 19:46,07 total
$ echo $?
0
```

### Full system restoration
A more real scenario as been successfully tested. A full system restoration of a Zimbra server instance within a container with high uid/gid.

So far, everything looks good. Thanks you sir!

**Comment by craigbarratt** [comment]
Thanks for the feedback, and thanks for doing some testing.  Please report back on the larger restore when it finishes.

**Comment by seblu** [comment]
The larger restore ended well. I think we can close here.
